### PR TITLE
wallet-api: CurrencySymbol is a bytestring

### PR DESCRIPTION
**Pull request by j-mueller**
* Add a builtin `emptyByteString :: SizedByteString 32` to `plutus-tx`.
* Change the type of `CurrencySymbol` to `SizedByteString 32`. For user-generated currencies this will be the hash of their monetary policy script (as per [the proposal](https://github.com/input-output-hk/plutus/blob/master/docs/multi-currency/multi-currency.md)). For Ada this will be the empty bytestring.
* Change `KeyBytes` to `LedgerBytes`. It wasn't used as a proper keybytes type (ie. a 64-byte bytestring consisting of a private and a public key) anyway. Now we just use it to define some `aeson` and `servant` instances to make our sized bytestrings easier to work with on the frontend.

@krisajenkins I wrote the instances and PS bridge for `CurrencySymbol` to behave like those for `Signature`. The PS code compiles and the tests pass, but it would be nice if you could have a look at it (JSON and PS both are a more verbose now).

### CODE REVIEW

File: plutus-tx/compiler/Language/PlutusTx/Compiler/Builtins.hs [hunk 1]
```diff
@@ -214,6 +218,10 @@ defineBuiltinTerms = do
         term <- wrapBsRel 2 $ mkBuiltin PLC.EqByteString
         defineBuiltinTerm 'Builtins.equalsByteString term [bs, bool]
 
+    do
+        let term = PIR.Constant () $ PLC.BuiltinBS () (fromIntegral $ natVal (Proxy @32)) BSL.empty
```
Comment (michaelpj) [inline plutus-tx/compiler/Language/PlutusTx/Compiler/Builtins.hs]: `natVal Proxy @32` is just `32`!

File: wallet-api/src/LedgerBytes.hs [hunk 2]
```diff
@@ -54,31 +54,38 @@ fromHex = KeyBytes . Builtins.SizedByteString . asBSLiteral
     asBSLiteral = withBytes asBytes
         where withBytes f = BSL.pack . f . BSL.unpack
 
-newtype KeyBytes = KeyBytes { getKeyBytes :: Builtins.SizedByteString 32 } -- TODO: use strict bytestring
+-- | 'Bultins.SizedByteString 32' with various useful JSON and
+--   servant instances for the Playground, and a convenient bridge
+--   type for PureScript.
+newtype LedgerBytes = LedgerBytes { getLedgerBytes :: Builtins.SizedByteString 32 } -- TODO: use strict bytestring
```
Comment (michaelpj) [inline wallet-api/src/LedgerBytes.hs]: This is kind of a weird type now but okay.

**Comment by j-mueller** [comment]
> I'm not entirely sure why we're using KeyBytes instead of just SizedByteString 32, though.

It's needed for the ToJSON, FromJSON, ToSchema, ToSchema instances that we use to represent hashes on the frontend. I didn't want to put the instances in `plutus-tx` because that project currently doesn't have any dependencies on `swagger`, `aeson` and `servant`. So we can use `SizedByteString` everywhere and use `LedgerBytes` only to derive those instances.

**Comment by j-mueller** [comment]
Test failure is a timeout on darwin: https://hydra.iohk.io/build/781457/nixlog/1 (built OK on linux)

**Pull request merged**